scraping/x/adajobo_scrapper.py

```diff
@@ -0,0 +1,372 @@
+                        continue
+                    else:
+                        bt.logging.error(
+                            f"Failed to run actor: {traceback.format_exc()}."
+                        )
+                        # This is an unfortunate situation. We have no way to distinguish a genuine failure from
+                        # one caused by malicious input. In my own testing I was able to make the Actor timeout by
+                        # using a bad URI. As such, we have to penalize the miner here. If we didn't they could
+                        # pass malicious input for chunks they don't have.
+                        return ValidationResult(
+                            is_valid=False,
+                            reason="Failed to run Actor. This can happen if the URI is invalid, or APIfy is having an issue.",
+                            content_size_bytes_validated=entity.content_size_bytes,
+                        )
+
+                # Parse the response
+                tweets = self._best_effort_parse_dataset(dataset)
+
+                actual_tweet = None
+                for tweet in tweets:
+                    if tweet.url == entity.uri:
+                        actual_tweet = tweet
+                        break
+                if actual_tweet is None:
+                    # Only append a failed result if on final attempt.
+                    if attempt == max_attempts:
+                        return ValidationResult(
+                            is_valid=False,
+                            reason="Tweet not found or is invalid.",
+                            content_size_bytes_validated=entity.content_size_bytes,
+                        )
+                else:
+                    return utils.validate_tweet_content(
+                        actual_tweet=actual_tweet,
+                        entity=entity,
+                    )
+
+        if not entities:
+            return []
+
+        # Since we are using the threading.semaphore we need to use it in a context outside of asyncio.
+        bt.logging.trace("Acquiring semaphore for concurrent adajobo validations.")
+        with AdajoboTwitterScraper.concurrent_validates_semaphore:
+            bt.logging.trace(
+                "Acquired semaphore for concurrent adajobo validations."
+            )
+            results = await asyncio.gather(
+                *[validate_entity(entity) for entity in entities]
+            )
+
+        return results
+
+    async def scrape(self, scrape_config: ScrapeConfig) -> List[DataEntity]:
+        """Scrapes a batch of Tweets according to the scrape config."""
+        # Construct the query string.
+        date_format = "%Y-%m-%d_%H:%M:%S_UTC"
+        bt.logging.debug(f"scrape conf: {scrape_config}")
+        query = f"since:{scrape_config.date_range.start.astimezone(tz=dt.timezone.utc).strftime(date_format)} until:{scrape_config.date_range.end.astimezone(tz=dt.timezone.utc).strftime(date_format)}"
```

Comment:
Does this format work on the new actor? Trying this manually returned no results for me: 
<img width="922" alt="Screenshot 2024-05-30 at 9 11 57 AM" src="https://github.com/RusticLuftig/data-universe/assets/151780164/4c1bdc06-f073-46f8-9f02-51d8ca85c2d2">

Reply:
Yes, it works